### PR TITLE
Dac6 409: Route users into Implementation date & Check answers page

### PR DESCRIPTION
**Pull request by ers2015112**
PR rebased off DAC6-417 so dependant on this being merged to master before merging

### CODE REVIEW

File: app/controllers/taxpayer/WhatIsTaxpayersStartDateForImplementingArrangementController.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+  private def getDisplayName(userAnswers: UserAnswers): Option[String] = {
+   userAnswers.get(TaxpayerSelectTypePage) map {
+     case Organisation => userAnswers.get(OrganisationNamePage) match {
+       case Some(organisationName) => organisationName
+       case None => "the organisation"
+     }
+     case Individual => userAnswers.get(IndividualNamePage) match {
+       case Some(individualName) => individualName.displayName
+       case None => "the individual"
```
Comment (helensoniega) [inline app/controllers/taxpayer/WhatIsTaxpayersStartDateForImplementingArrangementController.scala:117]: You can also call the getIndividualName and getOrganisationName in JourneyHelpers

File: app/utils/rows/TaxpayerRows.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import uk.gov.hmrc.viewmodels.SummaryList.{Action, Key, Row, Value}
+import uk.gov.hmrc.viewmodels.Text.Literal
+
+trait TaxpayerRows extends RowBuilder {
+
+  import pages.taxpayer.WhatIsTaxpayersStartDateForImplementingArrangementPage
+
+  def taxpayerSelectType: Option[Row] = userAnswers.get(TaxpayerSelectTypePage) map {
+    answer =>
+      Row(
```
Comment (helensoniega) [inline app/utils/rows/TaxpayerRows.scala:31]: Can be refactored to use toRow()

File: conf/views/taxpayer/selectType.njk [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            name: 'selectType',
+            fieldset: {
+              legend: {
+                text: messages('selectType.heading'),
+                isPageHeading: true,
+                classes: "govuk-fieldset__legend--xl"
+              }
+            },
+            items: radios,
+            errorMessage: form.confirm.error
```
Comment (helensoniega) [inline conf/views/taxpayer/selectType.njk:44]: Shouldn't this be form.selectType.error?

**Review by cameronstalker1 — CHANGES_REQUESTED** [review]

File: test/controllers/taxpayer/CheckYourAnswersTaxpayersControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    val templateCaptor = ArgumentCaptor.forClass(classOf[String])
+    val jsonCaptor = ArgumentCaptor.forClass(classOf[JsObject])
+
+    verify(mockRenderer, times(nrOfInvocations)).render(templateCaptor.capture(), jsonCaptor.capture())(any())
+
+    val json = jsonCaptor.getValue
+    val summaryRows = (json \ "taxpayersSummary").toString
+
+    println(s"\n\n\n$summaryRows\n\n\n")
```
Comment (helensoniega) [inline test/controllers/taxpayer/CheckYourAnswersTaxpayersControllerSpec.scala:59]: Remove println

**Review by helensoniega — APPROVED** [review]

**Pull request merged**